packages/site/cms-site/src/iframebridge/useBlockPreviewFetch.tsx

```diff
@@ -6,18 +6,20 @@ import { useIFrameBridge } from "./useIFrameBridge";
 
 const cachingFetch = createFetchInMemoryCache(fetch);
 
-export function useBlockPreviewFetch(graphqlApiUrl: string) {
-    const iFrameBridge = useIFrameBridge();
+export function useBlockPreviewFetch() {
+    const { showOnlyVisible, graphQLApiUrl } = useIFrameBridge();
+
+    if (!graphQLApiUrl) throw new Error("graphQLApiUrl not available");
```

Comment:
Could this lead to race conditions where the block data is already present, but the api url isn't?

Reply:
"Zwei Fliegen mit einer Klappe": https://github.com/vivid-planet/comet/pull/2826